src/components/timeout-panel/timeout-panel.js

```diff
@@ -0,0 +1,91 @@
+export default class Timeout {
+  constructor(context, time, url) {
+    this.context = context;
+    this.countDownSeconds = time;
+    this.urlOnZero = url;
+
+    this.panelContainer = context.parentElement;
+    this.panel = this.panelContainer.querySelector('.ons-js-panel-with-countdown');
+    this.countdown = this.panelContainer.querySelector('.ons-js-timeout-timer');
+    this.accessibleCountdown = this.panelContainer.querySelector('.ons-js-timeout-timer-acc');
+
+    // Language dependent text strings
+    this.minutesTextSingular = context.getAttribute('minutesTextSingular');
+    this.minutesTextPlural = context.getAttribute('minutesTextPlural');
+    this.secondsTextSingular = context.getAttribute('secondsTextSingular');
+    this.secondsTextPlural = context.getAttribute('secondsTextPlural');
+    this.countdownText = context.getAttribute('countdownText');
+    this.countdownExpiredText = context.getAttribute('countdownExpiredText');
+
+    // Settings
+    this.timers = [];
+    this.timerRunOnce = false;
+
+    // Start module
+    this.initialise();
+  }
+
+  async initialise() {
```

Comment:
Why are `initialise` and `startUiCountdown` `async` functions? the `await` on `runTimer` doesn't seem like it does anything because `runTimer` doesn't await on anything (nor return a custom Promise).

Reply:
@kruncher @rmccar Alot of this code has been lifted from another module which does similar things and those same functions require waiting for api responses. I think its probably worth a refactor here of the timeout-modal to put any reuseable code into its own module e.g. the ui countdown part so it can be imported into that and this module.

Reply:
I think it's also worth looking at the above comment I made about creating a separate module for the countdown that both the modal and panel can import.

Reply:
Sorry I misread that you said you would refactor.